src/Tribe/Timezones.php

```diff
@@ -0,0 +1,374 @@
+<?php
+
+/**
+ * Helpers for handling timezone based event datetimes.
+ *
+ * In our timezone logic, the term "local" refers to the locality of an event
+ * rather than the local WordPress timezone.
+ */
+class Tribe__Events__Timezones {
+	const SITE_TIMEZONE  = 'site';
+	const EVENT_TIMEZONE = 'event';
+
+
+	/**
+	 * Container for reusable DateTimeZone objects.
+	 *
+	 * @var array
+	 */
+	protected static $timezones = array();
+
+
+	public static function init() {
+		self::display_timezones();
+	}
+
+	/**
+	 * Takes care of appending timezone information to the display of
+	 * event date/times.
+	 */
+	protected static function display_timezones() {
+		if ( tribe_get_option( 'tribe_events_timezones_show_zone' ) ) {
+			add_filter( 'tribe_events_event_schedule_details', array( __CLASS__, 'append_timezone' ), 10, 2 );
```

Comment:
I was going to update the `tribe_events_event_schedule_details` hook so callbacks could access the before and after properties type-of-deal, rather than any CSS work.